src/core/server/http/http_config.ts

```diff
@@ -142,13 +143,18 @@ export class HttpConfig {
   public ssl: SslConfig;
   public compression: { enabled: boolean; referrerWhitelist?: string[] };
   public csp: ICspConfig;
+  public externalUrl: IExternalUrlConfig;
   public xsrf: { disableProtection: boolean; whitelist: string[] };
   public requestId: { allowFromAnyIp: boolean; ipAllowlist: string[] };
 
   /**
    * @internal
    */
-  constructor(rawHttpConfig: HttpConfigType, rawCspConfig: CspConfigType) {
+  constructor(
+    rawHttpConfig: HttpConfigType,
+    rawCspConfig: CspConfigType,
+    rawExternalUrlConfig: ExternalUrlConfig
+  ) {
```

Comment:
Why do we need to put this `externalUrl` config into the `http` one again? It's only used internally by the `rendering` service, right? It's to avoid creating a new service just to expose the config from it, right?

Reply:
> Why do we need to put this externalUrl config into the http one again? It's only used internally by the rendering service, right? It's to avoid creating a new service just to expose the config from it, right?

Yeah I was trying to take the path of least resistance here. It seemed like a logical place to put this service.

> Why don't we declare it as a separate service? 

We can if you'd prefer.

> Should we make the external_url service space-aware it introduces http dependency on SO / UiSettings. Which we must avoid

I didn't intend for this to be space-aware, and I don't see a need to do so in the near term.

Reply:
> I didn't intend for this to be space-aware, and I don't see a need to do so in the near term.

ok, I thought it's going to be a user-specific setting in the future, but I cannot find any references to such a requirement in  the parent issue.